src/positions.rs

```diff
@@ -224,39 +224,58 @@ impl<'a> fmt::Display for LocalRoomNameParseError<'a> {
     }
 }
 
+impl PartialEq<str> for LocalRoomName {
+    fn eq(&self, other: &str) -> bool {
+        let s = self.to_array_string();
+        s.eq_ignore_ascii_case(other)
+    }
+}
+impl PartialEq<LocalRoomName> for str {
+    #[inline]
+    fn eq(&self, other: &LocalRoomName) -> bool {
+        <LocalRoomName as PartialEq<str>>::eq(other, self)
```

Comment:
I'll impl `Eq`!

The reason I'm doing the explicit reference here is that we have all these other implementations for `&str`, `&&str`, etc., and I want to make sure that I don't end up creating an infinite loop. This one might be alright without it, or with some other annotations, but then the code could change in the future and it'd be super easy with all of these implementations to end up creating an infinitely recursive function. Will create a comment to give these motivations!

Reply:
Oh! I forgot - `Eq` can only be applied if the left and right sides are the same type. `LocalRoomName` does implement `Eq<LocalRoomName>`, but it can't implement it for `str`.

Reply:
Ah, I didn't know that. I suppose it makes sense 🤔